charts/signoz/templates/otel-collector/hpa.yaml

```diff
@@ -1,5 +1,5 @@
 {{- if and (.Values.otelCollector.autoscaling.enabled) (not .Values.otelCollector.autoscaling.keda.enabled) -}}
-apiVersion: autoscaling/v2beta2
+apiVersion: autoscaling/v2
```

Comment:
In case of K8s version `<1.23`, we would need it.

Can you please update it to this?

```suggestion
apiVersion: autoscaling/v2{{- if semverCompare "<1.23" .Capabilities.KubeVersion.Version }}beta2{{- end }}
```

Reply:
The PR has been updated to use `.Capabilities.APIVersions.Has` to detect if API is available.

Reply:
yes, that would work as well.